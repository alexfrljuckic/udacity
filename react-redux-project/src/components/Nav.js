import { Link } from "react-router-dom";
import { connect } from "react-redux";
import '../styling/Nav.css';
import { logoutUser } from "../actions/authedUser";

const Nav = ({ dispatch, authedUser, users }) => {
    const profilePic = users[authedUser]?.avatarURL;
    const userName = users[authedUser]?.name;

    const handleLogout = () => {
        dispatch(logoutUser());
    }
    return (
        <nav className="nav">
            <ul>
                    <li>
                        <Link to="/Home">Home</Link>
                    </li>
                    <li>
                        <Link to="/LeaderBoard" data-testid="LeaderBoard">LeaderBoard</Link>
                    </li>
                    <li>
                        <Link to="/new">New</Link>
                    </li>
                    <li>
                        <div className="user-detail">
                            <img className="profile-pic" src={profilePic} alt=""/>
                            <span data-testid="nav-name">{userName}</span>
                            
                        </div>
                        <Link to="/" onClick={() => handleLogout()}>Log Out</Link>
                        
                    </li>
            </ul>
        </nav>
    )
};

const mapStateToProps = ({ dispatch, authedUser, users }) => {
    return {
        dispatch,
        authedUser,
        users
    }
}
export default connect(mapStateToProps)(Nav);