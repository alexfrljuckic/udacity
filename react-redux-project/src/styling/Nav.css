.nav ul {
    display: flex;
    flex-direction: row;
    justify-content: flex-start;
    text-decoration: none;
  }
  
.nav li:first-child {
    padding-left: 0;
  }

.nav li:last-child {
    margin-left: 1300px;
    
  }

.user-detail {
  display: flex;
  align-items: center;
  border: whitesmoke 1px solid;
  border-radius: 10px;
  }

.profile-pic {
    height: 60px;
  }

li {
    list-style-type: none;
    padding: 10px;
    text-decoration: none;
  }

  nav {
    border-bottom: 2px solid gray;
  }