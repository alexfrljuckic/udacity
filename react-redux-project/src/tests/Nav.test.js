import { fireEvent, render, screen } from "@testing-library/react";
import '@testing-library/jest-dom'
import { Provider } from "react-redux";
import { legacy_createStore } from "redux";
import { MemoryRouter } from "react-router-dom";
import reducer from "../reducers";
import middleware from "../middleware";
import Nav from "../components/Nav";

const store = legacy_createStore(reducer,middleware);
describe('nav', () => {
    it('nav snapshot matches', () => {
        var cmp = render(
            <MemoryRouter>
                <Provider store={store}>
                        <Nav />
                </Provider>
            </MemoryRouter>
        )
        expect(cmp).toMatchSnapshot();
    })
})

describe('nav to tabs', () => {

    it('leaderboard', () => {
        render(
            <MemoryRouter>
                <Provider store={store}>
                        <Nav />
                </Provider>
            </MemoryRouter>
        )
        const leaderLink = screen.getByRole('link', { name : 'LeaderBoard' });
        expect(leaderLink).toHaveAttribute('href','/LeaderBoard')
    })

    it('new', () => {
        render(
            <MemoryRouter>
                <Provider store={store}>
                        <Nav />
                </Provider>
            </MemoryRouter>
        )
        const newLink = screen.getByRole('link', { name : 'New' });
        expect(newLink).toHaveAttribute('href','/new')
    })
})