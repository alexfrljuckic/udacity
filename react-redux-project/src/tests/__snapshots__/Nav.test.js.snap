// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nav nav snapshot matches 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="nav"
      >
        <ul>
          <li>
            <a
              href="/Home"
            >
              Home
            </a>
          </li>
          <li>
            <a
              data-testid="LeaderBoard"
              href="/LeaderBoard"
            >
              LeaderBoard
            </a>
          </li>
          <li>
            <a
              data-testid="newLink"
              href="/new"
            >
              New
            </a>
          </li>
          <li>
            <div
              class="user-detail"
            >
              <img
                alt=""
                class="profile-pic"
              />
              <span
                data-testid="nav-name"
              />
            </div>
            <a
              href="/"
            >
              Log Out
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="nav"
    >
      <ul>
        <li>
          <a
            href="/Home"
          >
            Home
          </a>
        </li>
        <li>
          <a
            data-testid="LeaderBoard"
            href="/LeaderBoard"
          >
            LeaderBoard
          </a>
        </li>
        <li>
          <a
            data-testid="newLink"
            href="/new"
          >
            New
          </a>
        </li>
        <li>
          <div
            class="user-detail"
          >
            <img
              alt=""
              class="profile-pic"
            />
            <span
              data-testid="nav-name"
            />
          </div>
          <a
            href="/"
          >
            Log Out
          </a>
        </li>
      </ul>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
